### PR TITLE
Redirect to details pages after creating task or campaign

### LINKED ISSUE #279: Navigate to newly created campaign when submitting on new campaign
See how this is done for tasks in `/src/components/ZetkinSpeedDial/actions/createTask.tsx`

### PR DESCRIPTION
**Pull request by omidhq**
Resolves #279

### CODE REVIEW

File: src/components/ZetkinSpeedDial/actions/createCampaign.tsx [hunk 1]
```diff
@@ -16,9 +16,12 @@ const DialogContent: React.FunctionComponent<DialogContentBaseProps> = ({ closeD
         onSettled: () => queryClient.invalidateQueries('campaigns'),
     });
 
-    const handleCreateCampaignFormSubmit = (data: Record<string,unknown>) => {
-        campaignMutation.mutate(data);
+    const handleCreateCampaignFormSubmit = async (data: Record<string,unknown>) => {
+        const newCampaign = await campaignMutation.mutateAsync(data);
         closeDialog();
```
Comment (nehahirve) [inline src/components/ZetkinSpeedDial/actions/createCampaign.tsx:21]: I wonder if it's necessary anymore to close the dialog if we're redirecting?

File: src/components/ZetkinSpeedDial/actions/createTask.tsx [hunk 2]
```diff
@@ -20,7 +20,7 @@ const DialogContent: React.FunctionComponent<DialogContentBaseProps> = ({ closeD
         closeDialog();
         const newTask = await eventMutation.mutateAsync(data);
         // Redirect to task page
-        router.push(`/organize/${orgId}/campaigns/${campId}/tasks/${newTask.id}`);
+        router.push(`/organize/${orgId}/campaigns/${campId}/calendar/tasks/${newTask.id}`);
```
Comment (nehahirve) [inline src/components/ZetkinSpeedDial/actions/createTask.tsx:23]: here too, do we need `closeDialog()` @richardolsson ? Here it's before we await the promise, in `createCampaign` it's after -- don't know what that's doing

File: src/fetching/postCampaign.ts [hunk 3]
```diff
@@ -12,6 +12,6 @@ export default function postCampaign(orgId : string, fetch = defaultFetch) {
             method: 'POST',
         });
         const resData = await res.json();
-        return resData;
+        return resData?.data;
```
Comment (nehahirve) [inline src/fetching/postCampaign.ts:15]: We should make a note of this in some cleanup issue that we're not returning from the fetch functions consistently -- there are some fetch functions that will throw an error, and some like this, that will return undefined if there is no `resData`

Comment (richardolsson) [reply src/fetching/postCampaign.ts:15]: Good idea!

@omidhq Can you create an issue for making return values from fetch functions consistent?

File: src/components/ZetkinSpeedDial/actions/createTask.tsx (hunk 2 above)
Comment (richardolsson) [reply src/components/ZetkinSpeedDial/actions/createTask.tsx:23]: Very valid point!

Lets get rid of the `closeDialog()` and check that everything is still working. Can you try that, @omidhq?

**Review by richardolsson — CHANGES_REQUESTED** [review]
See Changes suggested by @nehahirve and me in the discussions above.

File: src/components/ZetkinSpeedDial/actions/createTask.tsx (hunk 2 above)
Comment (omidhq) [reply src/components/ZetkinSpeedDial/actions/createTask.tsx:22]: Good point @nehahirve. It works perfectly without that function. Made the changes and pushed it again, @richardolsson.

File: src/fetching/postCampaign.ts (hunk 3 above)
Comment (omidhq) [reply src/fetching/postCampaign.ts:15]: @richardolsson Sure!

**Review by richardolsson — APPROVED** [review]
Looking good and working as expected. ✨ 

Great work on this! Go ahead and merge at will. 🟢

**Pull request merged**